core/src/ncipher/aes_gcm.c

```diff
@@ -26,6 +26,13 @@ uint8_t aes_gcm_buffer[1000];
 Result aes_gcm_encrypt(M_KeyID keyId, uint8_t *plaintext, size_t plaintext_len,
                        uint8_t *ciphertext, size_t ciphertext_len,
                        size_t *bytes_written) {
+
+  // NULL pointer checks. Note that plaintext can be NULL.
+  if (!ciphertext || !bytes_written) {
```

Comment:
nit: it's cleaner to break this into two different IF statements and provide a more accurate error message (exactly which variable was NULL).